LanguageFeatures/Patterns/type_inference_A17_t04.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Calculate the static type of the pattern. Using that value type,
+/// recurse through the pattern again downwards to the leaf subpatterns filling
+/// in any missing types in the pattern. This process may also insert implicit
+/// coercions and casts from dynamic when values flow into a pattern during
+/// matching.
+///
+/// @description Check the calculation static type of a list pattern. Test that
+/// missing types in a type schema are filled in from the initializing
+/// expression. Test the case when non-empty list pattern has context type
+/// schema _
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+import "../../Utils/static_type_helper.dart";
+
+main() {
+  var [...r1] = [1, 2, 3];
```

Comment:
We should have type arguments on the initializing expression here, as usual: If the type schema of the pattern is buggy and says `List<num>` then the initializing expression will be a `List<num>`, or the pattern could say `List<double>` and the elements of the initializing expression would be coerced to `double`s, etc.